epoxy-adapter/src/test/java/com/airbnb/epoxy/EpoxyControllerTest.java

```diff
@@ -0,0 +1,195 @@
+    };
+
+    controller.setFilterDuplicates(true);
+    controller.requestModelBuild();
+
+    assertEquals(1, controller.getAdapter().getItemCount());
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void throwOnDuplicatesIfNotFiltering() {
+    EpoxyController controller = new EpoxyController() {
+
+      @Override
+      protected void buildModels() {
+        new TestModel()
+            .id(1)
+            .addTo(this);
+
+        new TestModel()
+            .id(1)
+            .addTo(this);
+      }
+    };
+
+    controller.requestModelBuild();
+  }
+
+  boolean exceptionSwallowed;
+
+  @Test
+  public void exceptionSwallowedWhenDuplicateFiltered() {
+    EpoxyController controller = new EpoxyController() {
+
+      @Override
+      protected void buildModels() {
+        new TestModel()
+            .id(1)
+            .addTo(this);
+
+        new TestModel()
+            .id(1)
+            .addTo(this);
+      }
+
+      @Override
+      protected void onExceptionSwallowed(RuntimeException exception) {
+        exceptionSwallowed = true;
+      }
+    };
+
+    controller.setFilterDuplicates(true);
+    controller.requestModelBuild();
+
+    assertTrue(exceptionSwallowed);
+  }
+
+  boolean interceptorCalled;
+
+  @Test
+  public void interceptorRunsAfterBuildModels() {
+    EpoxyController controller = new EpoxyController() {
+
+      @Override
+      protected void buildModels() {
+        new TestModel()
+            .addTo(this);
+      }
+    };
+
+    controller.addInterceptor(new Interceptor() {
+      @Override
+      public void intercept(List<EpoxyModel<?>> models) {
+        assertEquals(1, models.size());
+        interceptorCalled = true;
+      }
+    });
+
+    controller.requestModelBuild();
+
+    assertTrue(interceptorCalled);
+    assertEquals(1, controller.getAdapter().getItemCount());
+  }
+
+  @Test
+  public void interceptorCanChangeModels() {
+    EpoxyController controller = new EpoxyController() {
+
+      @Override
+      protected void buildModels() {
+        new TestModel()
+            .addTo(this);
+      }
+    };
+
+    controller.addInterceptor(new Interceptor() {
+      @Override
+      public void intercept(List<EpoxyModel<?>> models) {
+        models.add(new TestModel());
+      }
+    });
+
+    controller.requestModelBuild();
+
+    assertEquals(2, controller.getAdapter().getItemCount());
+  }
+
+  List<EpoxyModel<?>> savedModels;
+
+  @Test(expected = IllegalStateException.class)
+  public void interceptorCannotAddModelsLater() {
+    EpoxyController controller = new EpoxyController() {
+
+      @Override
+      protected void buildModels() {
+        new TestModel()
+            .addTo(this);
+      }
+    };
+
+    controller.addInterceptor(new Interceptor() {
+      @Override
+      public void intercept(List<EpoxyModel<?>> models) {
+        savedModels = models;
+      }
+    });
+
+    controller.requestModelBuild();
+
+    savedModels.add(new TestModel());
+  }
```

Comment:
Maybe add a test with more than one interceptor, to verify that they're all executed, and in the right order.